Add hooks to disable open access and student previews

yscik opened the pull request:
Fixes #6303 

### Changes proposed in this Pull Request

* Add hooks described below

### Testing instructions

- Set a course to open access
- Add snippet `add_filter( 'sensei_feature_open_access_courses', '__return_false' );`
- Check that the open access course cannot be accessed as guest
- Check that the 'Open access' option is not visible in the course editor settings sidebar
- Replace previous snippet with `add_filter( 'sensei_course_open_access', '__return_false' );`
- Check that the 'Open access' option is visible, but the course is not accessible as guest even when its turned on.
--
- Add snippet `add_filter( 'sensei_feature_preview_students', '__return_false' );`
- Open a course frontend as admin/teacher
- Check that the 'Preview as student' link in the admin bar is not visible

<!-- Add the following only if there are new/updated actions or filters. Please provide a brief description of what they do and any arguments they may take. Be sure to also add the "Hooks" label to this PR. -->
### New/Updated Hooks

`sensei_feature_open_access_courses` (`() : bool`, `true` default) → When set to false, don't show the 'Open access' option in course settings, and don't allow guest users.

`sensei_course_open_access` (`( $course_id ) : bool`, setting from meta as default) → Override when checking if an individual course is open access.

`sensei_feature_preview_students` (`() : bool`, `true` default) → When set to false, don't show the 'Preview as Student' link in the admin bar, and disable all the preview user functions.

yscik committed:
Add hooks to disable open access and student previews

yscik committed:
Send option to editor

Imran92 reviewed (commented):
Works as expected for me, but clicking on Take Course on an Open Access course while having the `add_filter( 'sensei_course_open_access', '__return_false' );` set, didn't redirect me to log in page if I'm logged out. Can you kindly check if it happens to you too?

yscik committed:
Fix take course button conditions

yscik committed:
Merge branch 'feature/guest-user' into add/guest-preview-user-hooks

# Conflicts:
#	includes/class-sensei-guest-user.php

yscik commented:
Found the issue, fixing in 1a8267f. I moved around the checks on when to link to login or the start course action from the Take course button, so the Guest users class can filter `sensei_can_user_manually_enrol` instead of a direct meta check.

Also updated the legacy lesson action buttons that are displayed when Sensei blocks are not used — those would display Register instead of Take course when the user is logged out, but made that one show Take course too for open access courses:

<img width="267" alt="image" src="https://user-images.githubusercontent.com/176949/211893106-0cb810c8-01bc-489e-904c-404241dd285e.png"> <img width="252" alt="image" src="https://user-images.githubusercontent.com/176949/211893492-75bcf065-43e6-4dc2-a501-082b1fdab8bd.png">

Imran92 reviewed (approved):
Nicely done 👌 Works perfectly now
